pip/cmdoptions.py

```diff
@@ -523,6 +524,58 @@ def only_binary():
 )
 
 
+def _good_hashes():
+    """Return names of hashlib algorithms at least as strong as sha256."""
+    # Remove getattr when 2.6 dies.
+    algos = set(
+        getattr(hashlib,
+                'algorithms',
+                ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')))
+    return algos - set(['md5', 'sha1', 'sha224'])
```

Comment:
I don't think this is a good way to actually tell what the strength of these algorithms are. It assumes that there can never be an additional algorithm that isn't as strong as sha256 added to Python. Something like:

```python
import hashlib

def _good_hashes():
    return set([x for x in hashlib.algorithms if hashlib.new(x).digest_size >= 32])
```

Ideally without a magical 32 in there, for that matter this probably doesn't need to be a function at all does it?